src/java/org/apache/cassandra/utils/concurrent/SimpleCondition.java

```diff
@@ -101,4 +105,9 @@ public boolean awaitUntil(Date deadline) throws InterruptedException
     {
         throw new UnsupportedOperationException();
     }
+
+    public void reset() {
+        signaled = disabled;
+        waitingUpdater.set(this, null);
```

Comment:
`waitingUpdater.compareAndSet(this, null, new WaitQueue());` ?
+ throw if `waiting.hasWaiters()`